src/box/sql/mem.c

```diff
@@ -719,8 +719,7 @@ static inline int
 uint_to_dec(struct Mem *mem)
 {
 	assert(mem->type == MEM_TYPE_UINT);
-	int64_t u = mem->u.u;
-	decimal_from_uint64(&mem->u.d, u);
+	decimal_from_uint64(&mem->u.d, mem->u.u);
```

Comment:
AFAIR we agreed that `uint64_t -> int64_t -> uint64_t` conversions are safe on any sane arch and so we aren't fixing them (see the SVACE chat).